tools/generator/program.go

```diff
@@ -0,0 +1,424 @@
+		errLog.Print(err)
+		exitStatus = ExitCodeFailedToClone
+		return
+	}
+
+	if err = checkoutAPISpecsVer(azureRESTAPIBranch, repoLoc); err != nil {
+		errLog.Print(err)
+		exitStatus = ExitCodeFailedToClone
+		return
+	}
+
+	finderOutput := ioutil.Discard
+	if verbose {
+		finderOutput = os.Stdout
+	}
+
+	finder, err := NewSwaggerFinder(repoLoc, finderOutput)
+	if err != nil {
+		errLog.Print(err)
+		exitStatus = ExitCodeFinderFailure
+		return
+	}
+
+	if dryRun {
+		fmt.Println("Executing Preview")
+		namespaces := finder.Enumerate(nil).Select(func(x interface{}) interface{} {
+			namespace, err := getNamespace(x.(SwaggerFile))
+			if err != nil {
+				return err
+			}
+			return namespace
+		})
+
+		for namespace := range namespaces {
+			var output io.Writer
+			switch namespace.(type) {
+			case error:
+				output = os.Stderr
+			case string:
+				output = os.Stdout
+			}
+			fmt.Fprintln(output, namespace)
+		}
+		return
+	}
+
+	var logFolder string
+	useLogs := true
+
+	logFolder, err = ioutil.TempDir("", "autorestLogs")
+	if err != nil {
+		useLogs = false
+	}
+	fmt.Println("Logs will be stored in: ", logFolder)
+
+	found, generated, formatted, vetted := 0, 0, 0, 0
+
+	processor := finder.Enumerate(nil).Select(func(x interface{}) interface{} {
+		found++
+		return x
+	}).ParallelSelect(func(x interface{}) interface{} {
+		cast := x.(SwaggerFile)
+		var err error
+		var logWriter io.Writer
+		var outputFile *os.File
+		if useLogs {
+			outputFile, err = os.OpenFile(path.Join(logFolder, cast.Info.Title+"_"+cast.Info.Version+".txt"), os.O_WRONLY|os.O_CREATE, 0777)
+			logWriter = outputFile
+			if err == nil {
+				defer outputFile.Close()
+			} else {
+				fmt.Fprintln(os.Stderr, "Cannot log results of: ", cast.Info.Title, cast.Info.Version, "because: ", err)
+			}
+		} else {
+			logWriter = ioutil.Discard
+		}
+		var name string
+		name, err = generate(cast, outputLocation, repoLoc, logWriter)
+		if err != nil {
+			return err
+		}
+		generated++
+		return name
+	}).Where(func(x interface{}) bool {
+		switch x.(type) {
+		case string:
+			if verbose {
+				fmt.Fprintln(os.Stdout, "Generated: ", x)
+			}
+			return true
+		case error:
+			fmt.Fprintln(os.Stderr, "Error: ", x)
+		}
+		return false
+	}).Select(func(x interface{}) interface{} {
+		subject := x.(string)
+		subject = strings.TrimPrefix(subject, outputLocation)
+		subject = path.Join("github.com", "Azure", "azure-sdk-for-go", subject)
+		subject = path.Clean(subject)
+		return subject
+	}).Where(func(x interface{}) bool {
+		return format(x.(string)) == nil
+	}).Where(func(x interface{}) bool {
+		formatted++
+		return vet(x.(string)) == nil
+	})
+
+	for range processor {
+		vetted++
+	}
+
+	fmt.Println("Summary:")
+	fmt.Println("\tFound:    \t", found)
+	fmt.Println("\tGenerated:\t", generated)
+	fmt.Println("\tFormatted:\t", formatted)
+	fmt.Println("\tVetted:   \t", vetted)
+}
+
+func cloneAPISpecs(origin, local string) (string, error) {
+	_, cloneLoc := filepath.Split(local)
```

Comment:
Autorest does not need the repo to be cloned on local. You can do `autorest [https://raw.githubusercontent.com/Azure/azure-rest-api-specs/current/specification/graphrbac/data-plane/readme.md`](https://raw.githubusercontent.com/Azure/azure-rest-api-specs/current/specification/graphrbac/data-plane/readme.md%60).

Reply:
Ah, this is a new developement, that's great :)